src/Altinn.App.Core/Features/Validation/Default/ExpressionValidator.cs

```diff
@@ -102,29 +105,33 @@ IAppMetadata appMetadata
                             continue;
                         }
 
-                        var isInvalid = ExpressionEvaluator.EvaluateExpression(
+                        var validationResult = ExpressionEvaluator.EvaluateExpression(
                             evaluatorState,
-                            validation.Condition,
+                            validation.Condition.Value,
                             context,
                             positionalArguments
                         );
-                        if (isInvalid is not bool)
-                        {
-                            throw new ArgumentException(
-                                $"Validation condition for {resolvedField} did not evaluate to a boolean"
-                            );
-                        }
-                        if ((bool)isInvalid)
+                        switch (validationResult)
                         {
-                            var validationIssue = new ValidationIssue
-                            {
-                                Field = resolvedField,
-                                Severity = validation.Severity ?? ValidationIssueSeverity.Error,
-                                CustomTextKey = validation.Message,
-                                Code = validation.Message,
-                                Source = ValidationIssueSources.Expression,
-                            };
-                            validationIssues.Add(validationIssue);
+                            case true:
+                                var validationIssue = new ValidationIssue
+                                {
+                                    Field = resolvedField.Field,
+                                    DataElementId = resolvedField.DataType,
+                                    Severity = validation.Severity ?? ValidationIssueSeverity.Error,
+                                    CustomTextKey = validation.Message,
+                                    Code = validation.Message,
+                                    Source = ValidationIssueSources.Expression,
+                                };
+                                validationIssues.Add(validationIssue);
```

Comment:
The `DataElementId` should not be set to the data type id?

Reply:
No, that would not be very helpful. Good catch!